lib/vrl/parser/src/lex.rs

```diff
@@ -299,55 +302,55 @@ pub enum Token<S> {
 
 impl<S> Token<S> {
     pub(crate) fn map<R>(self, f: impl Fn(S) -> R) -> Token<R> {
-        use self::Token::*;
```

Comment:
Why this change? It seems unrelated, and personally, I dislike the extra noise it adds.

Reply:
We now have `Token::StringLiteral` and `Token::RawStringLiteral` enum variants. We also have  `StringLiteral` and `RawStringLiteral` structs.

By importing `self::Token::*` we get conflicts between these two types.

I can see that removing the `*` does add extra noise, but I fundamentally dislike wildcard imports because that noise does add extra information that I find useful, especially when there are potential name conflicts that arise from it.

If you feel strongly I can rename `StringLiteral` to `StringLiteralToken` to avoid the name conflict?

Reply:
I do feel strongly. Or, rather, I feel strongly that changing something that was chosen as a convention from the start of a project, needs to have good reasoning for why we want to move away from that convention.

"I fundamentally dislike X" doesn't cut it for me, I also fundamentally dislike certain stylistic choices in other parts of Vector, but I leave them as-is, as to (1) not clutter PRs, and (2) I know someone _liked_ those choices, since they picked them, so going my way, means also moving away from someone else's preference.

I dislike discussing these things, as it makes me come across as an annoying reviewer that debates minor issues, but I also feel like _not_ calling it out means it's free-for-all for people to change stylistic choices whenever they see something that they don't like, going back-and-forth in different PRs, so that's why I take the position of sticking with the choices made at the start of a project, unless there's a good reason (or consensus) to move away from those choices.

In this case, I can see why you did it, with the naming collision, but yeah, `StrLiteral` or `StringLiteralToken` would solve the issue 👍

Reply:
To be clear, I didn't mean I'm making the change just because I dislike wildcards, it's just the reason I dislike wildcards popped up and so I felt it was appropriate to remove it. But I'll make the change.